dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/chart/impl/DefaultChartService.java

```diff
@@ -796,7 +798,9 @@ else if ( chart.isAnalyticsType( AnalyticsType.EVENT ) )
         valueMap = DimensionalObjectUtils.getSortedKeysMap( valueMap );
 
         List<NameableObject> seriez = new ArrayList<>( chart.series() );
-        List<NameableObject> categories = new ArrayList<>( chart.category() );
+
+        List<DimensionalItemObject> dimensionalItemObjects = chart.category() != null ? chart.category() : emptyList();

```

Comment:
Do we know why this is null? I.e. in what circumstances is it null? I thought category should never be null.

Reply:
Some Charts like Pie or Gauge have only series. This chart generation is not prepared to support some kind of charts...